packages/arb-token-bridge-ui/src/hooks/useArbQueryParams.tsx

```diff
@@ -46,24 +46,39 @@ export const useArbQueryParams = () => {
 const isMax = (amount: string | undefined) =>
   amount?.toLowerCase() === AmountQueryParamEnum.MAX
 
-const sanitizeAmountQueryParam = (amountStr: string) => {
-  const amountNum = Number(amountStr)
-  // return original string if it is `max` (case-insensitive)
-  // if this check is removed, it'll satisfy the `isNaN` condition to return an empty string
-  if (isMax(amountStr)) {
-    return amountStr
+/**
+ * Sanitise amount value
+ * @param amount - transfer amount value from the input field or from the URL
+ * @returns sanitised value
+ */
+const sanitizeAmountQueryParam = (amount: string) => {
+  // no need to process empty string
+  if (amount.length === 0) {
+    return amount
   }
-  // to catch random string like `amount=asdf` from the URL
-  // to catch negative number
-  if (isNaN(amountNum) || amountStr?.length === 0) {
-    return ''
-  }
-  // for 0.0 / 0.00 ... that equals 0
-  if (amountNum === 0 && amountStr?.length > 0) {
-    return amountStr
+
+  // to catch strings like `amount=asdf` from the URL
+  if (isNaN(Number(amount))) {
+    // return original string if the string is `max` (case-insensitive)
+    // it doesn't show on the input[type=number] field because it isn't in the allowed chars
+    return isMax(amount) ? amount : ''
   }
+
   // to reach here they must be a number
-  return amountNum > 0 ? amountStr : String(Math.abs(amountNum))
+  // check for negative sign at first char
+  if (amount[0] === '-') {
```

Comment:
NIT: I'd use `amount.startsWith('-')` here.

Also, we can handle all cases with:
`Math.abs(parseFloat(amount)).toString()`

Reply:
the reason i don't use `Math.abs(Number(amount)).toString()` is because `0.0` will become `0` so the user cannot input `0.00000x` :'(
that was the bug

Reply:
i can move this condition `amount[0] === '.'` into the same if clause
but might be clearer if i just separate them? no strong preference

Reply:
Forgot about the `0.0` part. I would not group them, I think it's good like this.
No strong preference

Reply:
sounds good, updated